app/models/subject_rule.rb

```diff
@@ -13,7 +13,11 @@ class SubjectRule < ApplicationRecord
   }
 
   def condition
```

Comment:
I wonder if a lot of the code would have benefitted from this method not overriding the attribute accessor in the first place, so that `(rule.condition = condition) == rule.condition`. I regretted the name of this method later when I needed to work with these models in the Rails CLI as well.

Reply:
it's possible, but the `condition_string` nonsense is mainly there because nested arrays apparently aren't handled correctly by the way rails handles strong parameters